src/cacheRefresher.js

```diff
@@ -0,0 +1,185 @@
+/*
+Copyright 2022 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+(function () {
+    "use strict";
+
+    const { DomUtil } = require("./domUtil");
+    const XtkCaster = require('./xtkCaster.js').XtkCaster;
+    const { Cache } = require('./cache.js');
+
+    /**
+     * @private
+     * @class
+     * @constructor
+     * @memberof Campaign
+     */
+    class MetadataCache extends Cache {
+
+        /**
+         * A in-memory cache for properties values. Not intended to be used directly,
```

Comment:
Could you explain in more details what you mean by "properties values". The term "property" is very generic and not self explanatory. Giving examples of what properties are in this context can help. Also, maybe simply explaining that this is a cache used internally to store the state of the refresher. And by the way, why not calling this class RefresherStateCache to make it even more explicit?